test/test-proxy.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash -e
+
+base_url="https://localhost:9443/idp"
+client_id="oauth2-proxy"
+username="user2@konflux.dev"
+password="password"
+secrets_url="https://localhost:9443/api/k8s/workspaces/user-ns2/api/v1/namespaces/user-ns2/secrets"
+
+main() {
+    # First try without authorization, request should fail for 401 Authorization Required
+    echo "Get resource without authorization" >&2
+    secrets=$(curl -I -k -L -X GET "$secrets_url")
+    if ! echo "$secrets" | grep -q "401 Unauthorized"; then
+        echo "Error: Managed to reach resources without authorization"
+        exit 1
+    fi
+    echo "Success: Failed to get resource without authorization" >&2
+
+    # Get the client secret 
+    echo "Get client secret" >&2
+    base64_client_secret=$(kubectl get secrets -n dex oauth2-proxy-client-secret -o jsonpath='{.data.client-secret}')
+
+    # Decode it from base64
+    client_secret=$(printf '%s' "$base64_client_secret" | base64 -d)
+    
+    # Create auth header to send the request with, the format is base64 of <client_id>:<client_secret>
+    echo "Create header" >&2
+    auth_header=$(printf '%s' "$client_id:$client_secret" | base64)
+
+    # Send a request to get the token and extract the id_token
+    echo "Get token" >&2
+    id_token=$(curl -k -L -X POST "$base_url/token" \
+    -H "Authorization: Basic $auth_header" \
+    -H "Content-Type: application/x-www-form-urlencoded" \
+    --data-urlencode 'grant_type=password' \
+    --data-urlencode 'scope=openid profile email' \
+    --data-urlencode "username=$username" \
+    --data-urlencode "password=$password" | jq -r '.id_token' )
+
+    # Send a request for resources using the extracted id, request should succeed
+    echo "Get resource with authorization" >&2
+    secrets=$(curl -I -k -L -X GET "$secrets_url" -H "Authorization: Bearer $id_token")
```

Comment:
all the endpoints (location blocks) mentioned in the nginx config should be tested.
note that there are to location blocks that doesn't require auth (/ and /health)